.github/workflows/build.yml

```diff
@@ -0,0 +1,80 @@
+name: Build
+run-name: Build
+on:
+  workflow_dispatch:
+  push:
+    branches:
+    - master
+  pull_request:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+jobs:
+  test:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        python-version:
+        - 3.9.6
+        - 3.9.8
+        - 3.9.9
+        - 3.9.10
+        - 3.9.11
+        - 3.9.12
+        - 3.9.13
+        - 3.9.16
+        - 3.10.0
+        - 3.10.1
+        - 3.10.2
+        - 3.10.3
+        - 3.10.4
+        - 3.10.10
+    steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+        cache: 'pipenv'
```

Comment:
does the setup python actions provide caching for the dependencies ? Is that why we pass cache argument ?

Reply:
Yes, I pass cache argument there is to cache the pipenv dependencies. https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md